gems/activerecord/6.0/activerecord.rbs

```diff
@@ -289,6 +289,167 @@ module ActiveRecord
+      def eager_load: (*String | Symbol | Hash[untyped, untyped]) -> self
+      def preload: (*String | Symbol | Hash[untyped, untyped]) -> self
+      def find_by: (*untyped) -> Model?
+      def find_by!: (*untyped) -> Model
+      def find: (PrimaryKey id) -> Model
+              | (Array[PrimaryKey]) -> Array[Model]
+              | (*PrimaryKey) -> Array[Model]
+              | () { (Model) -> boolish } -> Model?
+      def find_or_create_by: (*untyped) -> Model
+                          | (*untyped) { (Model) -> void } -> Model
+      def find_or_create_by!: (*untyped) -> Model
+                            | (*untyped) { (Model) -> void } -> Model
+      def find_or_initialize_by: (*untyped) -> Model
+                              | (*untyped) { (Model) -> void } -> Model
+      def create_or_find_by: (*untyped) -> Model
+                          | (*untyped) { (Model) -> void } -> Model
+      def create_or_find_by!: (*untyped) -> Model
+                            | (*untyped) { (Model) -> void } -> Model
+      def take: () -> Model
+              | (Integer limit) -> Array[Model]
+      def take!: () -> Model
+      def first: () -> Model?
+              | (Integer count) -> Array[Model]
+      def first!: () -> Model
+      def second: () -> Model?
+      def second!: () -> Model
+      def third: () -> Model?
+      def third!: () -> Model
+      def fourth: () -> Model?
+      def fourth!: () -> Model
+      def fifth: () -> Model?
+      def fifth!: () -> Model
+      def forty_two: () -> Model?
+      def forty_two!: () -> Model
+      def second_to_last: () -> Model?
+      def second_to_last!: () -> Model
+      def third_to_last: () -> Model?
+      def third_to_last!: () -> Model
+      def last: () -> Model?
+              | (Integer count) -> Array[Model]
+      def last!: () -> Model
+      def limit: (Integer | Arel::Nodes::SqlLiteral | nil) -> self
+      def limit!: (Integer | Arel::Nodes::SqlLiteral | nil) -> self
+      def find_each: (?batch_size: Integer, ?start: Integer, ?finish: Integer, ?error_on_ignore: bool) { (Model) -> void } -> nil
+      def find_in_batches: (?batch_size: Integer, ?start: Integer, ?finish: Integer, ?error_on_ignore: bool) { (Array[Model]) -> void } -> nil
+      def in_batches: (?of: Integer, ?start: Integer, ?finish: Integer, ?load: bool, ?error_on_ignore: bool, ?order: untyped) { (self) -> void } -> nil
+      def destroy_all: () -> untyped
+      def delete_all: () -> untyped
+      def update_all: (*untyped) -> untyped
+      def touch_all: (*untyped, ?time: untyped) -> untyped
+      def destroy_by: (*untyped) -> untyped
+      def delete_by: (*untyped) -> untyped
+      def each: () { (Model) -> void } -> self
+      def select: (*Symbol | String) -> self
+                | () { (Model) -> boolish } -> Array[Model]
+      def reselect: (*Symbol | String) -> self
+    end
+
+    module ClassMethods[Model, Relation, PrimaryKey]
```

Comment:
Considering the future, it's a name that is likely to actually exist, so I think it would be better to have a name that is less likely to conflict.

Reply:
How about `ActiveRecord::Relation::ActiveRecord_Relation_ClassMethods`? It's very redundant, but maybe safety.